vme/src/experience.cpp

```diff
@@ -95,19 +115,29 @@ int shield_bonus(unit_data *att, unit_data *def, unit_data **pDef_shield)
         }
         else
         {
-            shield_bonus = OBJ_VALUE(def_shield, 1) + OBJ_VALUE(def_shield, 2);
+            shield_bonus = OBJ_VALUE(def_shield, 1) + OBJ_VALUE(def_shield, 2);*/
+/*This code gets shield bonus values and is equivalent to the numbers above but takes into account the shield type itself*/
+        shield_bonus = OBJ_VALUE(def_shield, 0) + OBJ_VALUE(def_shield, 1) + OBJ_VALUE(def_shield, 2); 
+        att_val += effective_dex(att); // This is equivalent to the above check to get att_dex
+/*This section switches between useing the minimum of Str, Con, and effective dex for a tower shield or effective dex for defending with the shield. Roughly equivalent to def_dex = effective_dex(def) commented out above*/
+        if (OBJ_VALUE(def_shield, 0) == SHIELD_TOWER){
+            def_val += MIN(MIN(CHAR_STR(def),(CHAR_CON(def))),effective_dex(def));
+        }else{
+             def_val += effective_dex(def);
+        }
 
             /* Let's make a shield check - CAN_SEE does affect this too */
-            hm = resistance_skill_check(def_dex + shield_bonus,
-                                        att_dex,
-                                        def->isPC() ? PC_SKI_SKILL(def, SKI_SHIELD) : def_dex,
-                                        att->isPC() ? PC_SKI_SKILL(att, SKI_SHIELD) : att_dex);
-
+            hm = resistance_skill_check(def_val + shield_bonus,
+                                        att_val,
+                                        def->isPC() ? PC_SKI_SKILL(def, SKI_SHIELD) : def_val,
+                                        att->isPC() ? PC_SKI_SKILL(att, SKI_SHIELD) : def_val);
+        
             if (hm >= 0)
-            { /* Successful Shield use */
-                def_shield_bonus = g_shi_info[OBJ_VALUE(def_shield, 0)].melee + shield_bonus / 2;
+            { /* Successful Shield use
+                 Modified to make who well you succeeded matter for how well you blocked*/
+                def_shield_bonus = (hm/10)+g_shi_info[OBJ_VALUE(def_shield, 0)].melee + shield_bonus / 2;
```

Comment:
Here OBJ_VAKUE(0) is used correctly via the g_shi_info as far as I can tell. 

The main change here is adding in (hm/10).  I suppose it's worth a try, but if it's too beneficial, we can just drop it again.